xml/System.Diagnostics/DiagnosticListener.xml

```diff
@@ -279,9 +279,19 @@
         <Parameter Name="isEnabled" Type="System.Func&lt;System.String,System.Object,System.Object,System.Boolean&gt;" Index="1" FrameworkAlternate="netcore-2.0;netcore-2.1;netcore-2.2;netcore-3.0" />
       </Parameters>
       <Docs>
-        <param name="observer">To be added.</param>
-        <param name="isEnabled">To be added.</param>
-        <summary>To be added.</summary>
+        <param name="observer">An <see cref="T:System.IObserver" /> subscriber.</param>
+        <param name="isEnabled">Filters events based on their name (<see cref="T:System.String" />) and up to two context object (which can be <see langword="null" />).  
+            
+A particular instrumentation site has the option of calling one or more <see cref="M:System.Diagnostics.DiagnosticListener.IsEnabled" /> overloads  in which it passes the name of the event and up to two other (instrumentation site specific) objects as arguments. If any of these <see cref="M:System.Diagnostics.DiagnosticListener.IsEnabled" /> calls are made then this <paramref name="isEnabled" /> predicate is invoked with passed values (if shorter overloads are used, <see langword="null" /> is passed for missing context objects).
+            
+This gives any particular instrumentation site the ability to pass up to two pieces of information to the subscriber to do sophisticated, efficient filtering. This requires more coupling between the instrumentation site and the subscriber code.
+            
+It is expected that a particular instrumentation site may call different overloads of IsEnabled for the same event, first calling <see cref="M:System.Diagnostics.DiagnosticListener.IsEnabled(System.String)" /> which calls the filter with two <see langword="null" /> context objects and if <paramref name="isEnabled" /> returns <see langword="true" /> calling again with context objects. The <paramref name="isEnabled" /> filter should be designed with this in mind. 
```

Comment:
This sentence is confusing/run-on - maybe try:

> first calling IsEnabled(System.String), which calls the filter with two null context objects. If isEnabled returns true, it calls again with context objects.

Reply:
```suggestion
It is expected that a particular instrumentation site may call different overloads of IsEnabled for the same event, first calling <see cref="M:System.Diagnostics.DiagnosticListener.IsEnabled(System.String)" /> which calls the filter with two <see langword="null" /> context objects. If <paramref name="isEnabled" /> returns <see langword="true" />, it calls again with context objects. The <paramref name="isEnabled" /> filter should be designed with this in mind. 
```